boot/mbed/mbed_lib.json

```diff
@@ -0,0 +1,159 @@
+            "value": 4096
+        },
+        "validate-primary-slot": {
+            "help": "Always check the signature of the image in the primary slot before booting, even if no upgrade was performed. This is recommended if the boot time penalty is acceptable.",
+            "macro_name": "MCUBOOT_VALIDATE_PRIMARY_SLOT",
+            "accepted_values": [true, null],
+            "value": true
+        },
+        "signature-algorithm": {
+            "help": "The algorithm used for digital signing.",
+            "macro_name": "MCUBOOT_SIGNATURE_ALGORITHM",
+            "required": true,
+            "accepted_values": ["SIGNATURE_TYPE_RSA", "SIGNATURE_TYPE_EC256", "SIGNATURE_TYPE_ED25519"],
+            "value": "SIGNATURE_TYPE_RSA"
+        },
+        "rsa-signature-length": {
+            "help": "If RSA is used for signature algorithm, this specifies the length.",
+            "macro_name": "MCUBOOT_RSA_SIGNATURE_LENGTH",
+            "required": true,
+            "accepted_values": [2048, 3072],
+            "value": 2048
+        },
+        "crypto-backend": {
+            "help": "The crypto library backend. NOTE: TinyCrypt is currently unsupported for Mbed OS, until we implement TinyCrypt's RNG interface",
+            "macro_name": "MCUBOOT_CRYPTO_BACKEND",
+            "required": true,
+            "accepted_values": ["MBEDTLS", "TYNYCRYPT"],
+            "value": "MBEDTLS"
+        },
+        "overwrite-only": {
+            "help": "The default is to support A/B image swapping with rollback. A simpler code path, which only supports overwriting the existing image with the update image, is also available. (null to disable)",
+            "macro_name": "MCUBOOT_OVERWRITE_ONLY",
+            "accepted_values": [true, null],
+            "value": true
+        },
+        "overwrite-only-fast": {
+            "help": "Only erase and overwrite those primary slot sectors needed to install the new image, rather than the entire image slot",
+            "macro_name": "MCUBOOT_OVERWRITE_ONLY_FAST",
+            "accepted_values": [true, null],
+            "value": null
+        },
+        "log-level": {
+            "help": "Verbosity of MCUboot logging",
+            "macro_name": "MCUBOOT_LOG_LEVEL",
+            "accepted_values": ["MCUBOOT_LOG_LEVEL_OFF", "MCUBOOT_LOG_LEVEL_ERROR", "TRACE_LEVEL_WARN", "MCUBOOT_LOG_LEVEL_INFO", "MCUBOOT_LOG_LEVEL_DEBUG"],
+            "value": "MCUBOOT_LOG_LEVEL_OFF"
+        },
+        "max-img-sectors": {
+            "help": "Maximum number of flash sectors per image slot. Target-dependent, please set on a per-target basis",
+            "macro_name": "MCUBOOT_MAX_IMG_SECTORS",
+            "required": true
+        },
+        "read-granularity": {
+            "help": "Granularity of read operations, in bytes. It should be no less than the read size of all storage devices (no need to be multiples of them).",
+            "macro_name": "MCUBOOT_READ_GRANULARITY",
+            "value": null
```

Comment:
You mentioned that MCUboot requires a granularity of 16 bytes.
I think you need this when using a secondary block device from QSPIF. Can you explain how you use this configuration?

Reply:
Do you mean explain it in "help" here?

Reply:
I meant in a reply comment in the PR. Or perhaps I should review the applications related PR.

Reply:
It's used in `flash_area_read()` - it creates a buffer `uint8_t buffer[MCUBOOT_READ_GRANULARITY]`, reads data into the buffer, then memcpy's whatever is needed to the input buffer.